blueprints/addon/files/tsconfig.declarations.json

```
@@ -0,0 +1,10 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "declarationDir": "declarations",
+    "emitDeclarationOnly": true,
+    "isolatedModules": false,
```

```suggestion
    "rootDir": ".",
```

The `isolatedModules: false` flag that we had in `ts:precompile` was specifically to work around some incompatibilities with other flags that we needed to set in order to understand the structure of the output. Since we don't have tooling that's having to tweak flags any more and users can change their config however they like, forcing `isolatedModules` off should be unnecessary.

On the other hand, `tsc` can be a little cute about how it lays out its output: even though you have multiple directories in `include` below, if it only actually discovers `.ts` files in one of them, it will implicitly treat that as the input root. In other words, if the only TS file in your addon is `addon/index.ts`, it will generate `declarations/index.d.ts` without the intermediate `addon` directory. Explicitly setting `rootDir` here should ensure it always reflects the input structure appropriately.